src/backend/columnar/sql/columnar--10.0-3--10.1-1.sql

```diff
@@ -21,6 +21,30 @@ END$proc$;
 -- columnar objects when upgrading postgres
 DROP FUNCTION citus_internal.columnar_ensure_objects_exist();
 
+-- For a proper mapping between tid & (stripe, row_num), add a new column to
+-- columnar.stripe and define a BTREE index on this column.
+-- Also include storage_id column for per-relation scans.
+ALTER TABLE columnar.stripe ADD COLUMN first_row_number bigint;
+CREATE INDEX stripe_first_row_number_idx ON columnar.stripe USING BTREE(storage_id, first_row_number);
+
+-- Populate first_row_number column of columnar.stripe table.
+--
+-- For simplicity, we calculate MAX(row_count) value across all the stripes
+-- of all the columanar tables and then use it to populate first_row_number
```

Comment:
in `upgrade_columnar_metapage_before.sql`, set stripe row limit to different values for some tables, and have a test to show that we consider `MAX(row_count)` for all tables